### PR TITLE
Chat frontend

### PR DESCRIPTION
**Pull request by HelenBassa**
chat-frontend1
chat-frontend2
chat-frontend3

***Chat frontend***
In scope:

- [x] client should store all the room session messages
- [x] new messages should be displayed in the chat component
- [x] client's messages should be highlighted
- [x] should handle the new message from the client

### CODE REVIEW

File: client/src/components/ui/chat-component/message-input-button/message-input-button.tsx [hunk 1]
```diff
@@ -1,31 +1,55 @@
...
-  const inputHandler = (e: ChangeEvent<HTMLInputElement>): void => {
+  const roomID: string = useSelector((state: RootState) => state.game.room);
+  const clientUser: IUser = useSelector((state: RootState) => state.client.clientUser);
+  const ws = useContext(WebSocketContext);
+
+  const onChangeMessageHandler = (e: ChangeEvent<HTMLInputElement>): void => {
     if (e.target.value !== 'Enter') {
-      setInputValue(e.target.value);
+      setMessageText(e.target.value);
       console.log('set value', e.target.value);
```
Comment (utrennik) [inline client/src/components/ui/chat-component/message-input-button/message-input-button.tsx]: You can remove all the console.logs in the chat component if you dont need them anymore

File: client/src/components/ui/chat-component/message-list/message-list.tsx [hunk 2]
```diff
@@ -16,24 +26,48 @@ const MessageList = () => {
...
-            lastName={lastName}
-            avatarImage={avatarImage}
-            message={message}
-            isCurrentUser={isCurrentUser}
-          />
-        );
-      })}
+      {messages.map(
+        (msg) =>
+          (dealer.id === msg.userID && dealer.room === msg.room && (
```
Comment (utrennik) [inline client/src/components/ui/chat-component/message-list/message-list.tsx]: Why do we need this double-check? Are we receiving all the messages from all chat rooms?
I think, you want here to highlight the message from the clientUser, we have his data in the Redux store, you can use it.

Comment (utrennik) [reply client/src/components/ui/chat-component/message-list/message-list.tsx]: Maybe we can map through the received messages array and display them in the time-added order, highlighting depending on the current clientUser?

File: client/src/components/ui/chat-component/message-list/message-list.tsx [hunk 3]
```diff
@@ -16,24 +26,48 @@ const MessageList = () => {
...
+              key={msg.messageID}
+              messageID={msg.messageID}
+              room={msg.room}
+              userID={dealer.id}
+              isLobby={isLobby}
+              firstName={dealer.firstName}
+              lastName={dealer.lastName}
+              avatarImage={dealer.avatar}
+              message={msg.message}
+              isCurrentUser={client.id === dealer.id}
```
Comment (aksiantsiuk) [inline client/src/components/ui/chat-component/message-list/message-list.tsx]:
```js
const isCurrentUser = client.id === dealer.id
isCurrentUser={isCurrentUser}
```

File: client/src/components/ui/chat-component/message-list/message-list.tsx [hunk 4]
```diff
@@ -16,24 +26,48 @@ const MessageList = () => {
...
+            />
+          )) ||
+          users.map(
+            (user) =>
+              user.id === msg.userID &&
+              user.room === msg.room && (
```
Comment (aksiantsiuk) [inline client/src/components/ui/chat-component/message-list/message-list.tsx]: Set an understandable boolean variable

File: client/src/components/ui/chat-component/message-list/message-list.tsx [hunk 5]
```diff
@@ -16,24 +26,48 @@ const MessageList = () => {
...
+                  key={msg.messageID}
+                  messageID={msg.messageID}
+                  room={msg.room}
+                  userID={user.id}
+                  isLobby={isLobby}
+                  firstName={user.firstName}
+                  lastName={user.lastName}
+                  avatarImage={user.avatar}
+                  message={msg.message}
+                  isCurrentUser={client.id === user.id}
```
Comment (aksiantsiuk) [inline client/src/components/ui/chat-component/message-list/message-list.tsx]: the same as above

File: client/src/redux/reducers/client-reducer.ts [hunk 6]
```diff
@@ -1,4 +1,4 @@
-import config from '@src/config.json';
+// import config from '@src/config.json';
```
Comment (aksiantsiuk) [inline client/src/redux/reducers/client-reducer.ts]: remove the line if it useless

**Review by RealOneReal — APPROVED** [review]

File: client/src/components/ui/chat-component/message-input-button/message-input-button.tsx (hunk 1 above)
Comment (HelenBassa) [reply client/src/components/ui/chat-component/message-input-button/message-input-button.tsx]: Thanks, fixed

File: client/src/components/ui/chat-component/message-list/message-list.tsx (hunk 2 above)
Comment (HelenBassa) [reply client/src/components/ui/chat-component/message-list/message-list.tsx]: Thanks, fixed

File: client/src/components/ui/chat-component/message-list/message-list.tsx (hunk 3 above)
Comment (HelenBassa) [reply client/src/components/ui/chat-component/message-list/message-list.tsx]: Thanks, fixed

File: client/src/components/ui/chat-component/message-list/message-list.tsx (hunk 4 above)
Comment (HelenBassa) [reply client/src/components/ui/chat-component/message-list/message-list.tsx]: Thanks, fixed

File: client/src/components/ui/chat-component/message-list/message-list.tsx (hunk 5 above)
Comment (HelenBassa) [reply client/src/components/ui/chat-component/message-list/message-list.tsx]: Thanks, fixed

File: client/src/redux/reducers/client-reducer.ts (hunk 6 above)
Comment (HelenBassa) [reply client/src/redux/reducers/client-reducer.ts]: Thanks, fixed

**Review by utrennik — APPROVED** [review]

**Review by aksiantsiuk — APPROVED** [review]

**Pull request merged**